tutoraspects/templates/openedx-assets/queries/fact_course_grades.sql

```diff
@@ -3,10 +3,10 @@ with grades as (
     from {{ DBT_PROFILE_TARGET_DATABASE }}.fact_grades
     where grade_type = 'course'
     {% raw %}
-    {% if filter_values('course_name') != [] %}
+    {% if get_filters('course_name', remove_filter=True) != [] && filter_values('course_name') != [] %}
```

Comment:
My understanding of `remove_filter` is that it prevents Superset from adding the filter to the final generated SQL. I've used it in the past when we want to filter on a column in a subquery but don't want to include that column in the final result. What's not clear to me is how this would affect subsequent calls to `filter_values('course_name')` within this query. 

Additionally, the `get_filters` check feels redundant; I can't tell if there will be a case where that returns a different answer than the comparison on `filter_values('course_name')`.

Reply:
It shouldn't be in the same condition, I was testing some changes. The final SQL should be:

```jinja
{% if get_filters('course_name', remove_filter=True) != [] %}
# When there are no filters defined (such as in the SQL Lab, programmatic access, or when creating charts)
# Do nothing, no filter
{% if filter_values('course_name') != [] %}
# Bassically in the dashboard context
# Do the filter thing
{% else %}
 1=0
{% endif %}

```

Reply:
`get_filters` returns the applied filters to a column in a list with the comparator and the value. See https://superset.apache.org/docs/installation/sql-templating/#available-macros